lib/posix-process/signal/deliver.c

```diff
@@ -20,8 +21,16 @@ void pprocess_signal_call_handler_with_stack(int signum, siginfo_t *si,
 
 static void uk_sigact_term(int __unused sig)
 {
-	uk_pr_warn("tid %d terminated by signal\n", uk_sys_gettid());
-	pprocess_kill(uk_pprocess_current());
+	struct posix_process *pprocess;
+
+	pprocess = uk_pprocess_current();
+	UK_ASSERT(pprocess);
+
+	uk_pr_info("pid: %d terminated by signal\n", pprocess->pid);
+
+	pprocess_exit(pprocess, POSIX_PROCESS_KILLED, sig);
+	uk_sched_yield();
```

Comment:
Correct me if I am wrong: `pprocess_exit` calls `pprocess_exit_pthread` for each of its pthreads (including current one) which calls `uk_sched_thread_terminate` for all threads **BUT** the current one. Am I missing something or could this thread actually return from `uk_sched_yield`? What stops it from not doing so.

Reply:
That's right, it's a leftover from a previous iteration that I forgot to replace with `uk_sched_thread_exit()`. Will fix 👍🏼